internal/expr/prepare.go

```diff
@@ -0,0 +1,73 @@
+package expr
+
+import (
+	"bytes"
+	"fmt"
+)
+
+// PreparedExpr contains an SQL expression that is ready for execution.
+type PreparedExpr struct {
+	ParsedExpr *ParsedExpr
+	SQL        string
+}
+
+type typeNameToInfo map[string]*info
+
+// prepareInput checks that the input expression corresponds to a known type.
+func prepareInput(ti typeNameToInfo, p *inputPart) error {
+	inf, ok := ti[p.source.prefix]
+	if !ok {
+		return fmt.Errorf(`unknown type: "%s"`, p.source.prefix)
+	}
+
+	if _, ok = inf.tagToField[p.source.name]; !ok {
+		return fmt.Errorf(`no tag with name "%s" in "%s"`,
```

Comment:
Similarly, maybe:
```
cannot prepare expression: type Fooz has no "field_name" db tag
```
The field name here needs to be quoted because being a word it could be mixed with the message and misinterpreted.

Reply:
Changed.